README.md

```diff
@@ -1,122 +1,241 @@
+Other means of synchronizing the content (e.g. rsync, scp, or possibly even file system
+replication tools like GlusterFS) are not RRDP aware and thus may sync the Notification File
+before the files it references have all been downloaded which can cause Relying Party clients
+of the public facing repository servers to encounter errors while attempting to download updates
+from the service.
+
+## Synchronization
 
-The simplest use case (assuming that the invoking user has write access to the default directories, see below) requires only the URL where the RFC 8182 RRDP Notification File can be downloaded, e.g.:
+When `krill-sync` runs without previous state, it will retrieve and parse the specified
+notification file. Then it will retrieve and parse the snapshot file to obtain all current
+objects, as well as each delta file mentioned (unless an optional hard limit is set, see below).
 
+If a file cannot be retrieved or **parsed** then the run will fail. This means that if the
+back-end system RFC 8182 RRDP XML is somehow not understood by krill-sync the data will
+not be accepted. We believe that this should not happen because of our testing, so this should
+be considered a sanity check, and a feature, rather than a bug. But, of course if you run
+into issues with this please let us know!
+
+The tool will persist its current state to disk. The default directory for this is
+`/var/lib/krill-sync` but this can be overridden using the `--state-dir` argument.
+
+On subsequent runs `krill-sync` will check whether delta files can be used to update its
+current state. I.e. the session is unchanged and there is a chain of delta files
+available. If not, it will re-sync using the current back-end snapshot.
+
+**TODO:**
+1. Fall back to snapshot if a delta is not available
+2. Resync if there is regression in the back-end serial (without session reset)?
+
+
+## Produced RRDP Data
+
+This tool produces its own RRDP data XML. This means that the formatting, hash values
+and even the URI paths of the RRDP files produced by this tool may be **different**
+from the back-end. This should not be any issue as long as all your public facing
+RRDP front-ends use `krill-sync`.
+
+The RRDP files are written in a safe order, meaning: new snapshot and delta files are
+written to disk first. Then a new notification file is written to a temporary file,
+which is then renamed to avoid avoid race conditions resulting from overwriting this
+file as it's being served.
+
+Old snapshot and delta files are marked as 'deprecated' as soon they are no longer referenced
+by the current notification file. Files which have been deprecated for longer than N seconds
+are removed at the end of each synchronization (default 10 minutes).
```

Comment:
Synchronisation runs do not default to 10 minute intervals as krill-sync does not have a daemon mode yet, right? So perhaps (default 10 minutes) should go after "N seconds" ?

Reply:
revised